utils/zerovec/src/lib.rs

```diff
@@ -523,6 +523,53 @@ pub use zerovec_derive::make_varule;
 mod tests {
     use super::*;
     use core::mem::size_of;
+    use schemars::{gen::SchemaGenerator, JsonSchema};
+    use serde::{Deserialize, Serialize};
+    use serde_json::Value;
+
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    #[derive(JsonSchema)]
+    pub struct DataStruct<'data> {
```

Comment:
Suggestion: for in-file unit tests, make one test for each of these. For example,

```rust
#[test]
fn test_schema_zerovec_u32() {
    let gen = SchemaGenerator::default();
    let schema = gen.into_root_schema_for::<ZeroVec<u32>>();
    // ...
}
```

As suggested below, use insta for the more complex tests.